Transit machine_restart to the CPU0

dsemenets opened the pull request.

klogg commented:
I do not quite understand the description

lorc commented:
> I do not quite understand the description

As do I. This patch looks like work around some deeper issue. But this is not described, so it is unclear why we need to issue `machine_restart` only on CPU0.

dsemenets committed:
xen: Don't call panic if ARM TF cpu off returns DENIED

According to PSCI specification ARM TF can return DENIED on CPU OFF.
This patch brings the hypervisor into compliance with the PSCI
specification.
Refer to "Arm Power State Coordination Interface (DEN0022D.b)"
section 5.5.2

Signed-off-by: Dmytro Semenets <dmytro_semenets@epam.com>
Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

lorc commented on xen/arch/arm/psci.c:
```diff
@@ -63,8 +63,9 @@ void call_psci_cpu_off(void)
         arm_smccc_smc(PSCI_0_2_FN32_CPU_OFF, &res);
-        panic("PSCI cpu off failed for CPU%d err=%d\n", smp_processor_id(),
-              PSCI_RET(res));
+        if (PSCI_RET(res) != PSCI_DENIED)
```
Please follow Xen coding style:
`        if ( PSCI_RET(res) != PSCI_DENIED )`

lorc reviewed (commented):
It will be good to mention exact specification and locations.

Hint: this is "Arm Power State Coordination Interface (DEN0022D.b)" section 5.5.2

lorc commented:
With both comments fixed:

`Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`
